documentation/CAMARA-ICM-examples.md

```diff
@@ -0,0 +1,188 @@
+# Identity and Consent Management Examples
+
+## Specifying One Purpose
+
+---
+**Note**
+
+Access tokens content or structure are not part of the OAuth2 nor the OIDC standard. In [RFC7662](https://datatracker.ietf.org/doc/html/rfc7662) only the field `active` is REQUIRED.
+`scope` and all other fields are optional. [JSON Web Token](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1) defines some common claims.
+RFC7662 response values serve as an **example** how an access token might look like. These access tokens might contain additional fields carrying what Camara needs regarding "purpose"
+
+The scope `openid` is needed only in the request to specify that the request is an OpenId request. The scope `openid` is not needed in the access token.
+
+---
+**Note**
+
+This document uses the response of the token-introspection endpoint as per RFC7662 to describe an access token.
+This document does not say that the access token is self-contained or not.
+
+---
+
+### Purpose as a scope: Requesting one purpose, two scopes of the same API
+
+#### OIDC authorization code flow with one purpose as scope
+
+```
+GET /authorize?
+    response_type=code
+    &scope=openid%20dpv%3AFraudPreventionAndDetection%20sim-swap%3Acheck%20sim-swap%3Aretrieve-date
+    &client_id=s6BhdRkqt3
+    &state=af0ifjsldkj
+    &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb HTTP/1.1
+Host: server.example.com
+```
+
+#### RFC9101 request object with one purpose as scope
+
+```
+{
+  "iss": "s6BhdRkqt3",
+  "aud": "https://server.example.com",
+  "response_type": "code",
+  "client_id": "s6BhdRkqt3",
+  "redirect_uri": "https://client.example.org/cb",
+  "scope": "openid dpv:FraudPreventionAndDetection sim-swap:check sim-swap:retrieve-date",
+  "state": "af0ifjsldkj",
```

Comment:
```suggestion
  "sub" : "Z5O3upPC88QrAjx00dis",
  "state": "af0ifjsldkj",
```
sub claim is also part of access token

Reply:
I think this suggestion is stale. Marking this conversation as resolved.